File: src/main/java/me/adrigamer2950/adriapi/api/library/manager/LibraryManagerImpl.java
package me.adrigamer2950.adriapi.api.library.manager;

import net.byteflux.libby.BukkitLibraryManager;
import net.byteflux.libby.Library;
import org.bukkit.plugin.Plugin;

import java.util.Collection;
import java.util.HashSet;
import java.util.Set;

public class LibraryManagerImpl extends BukkitLibraryManager implements LibraryManager {

    protected final Plugin plugin;
    protected final Set<Library> libraries = new HashSet<>();

    public LibraryManagerImpl(Plugin plugin) {
        super(plugin);

        this.plugin = plugin;

        this.addMavenCentral();
    }

    @Override
    public Plugin getPlugin() {
        return this.plugin;
    }

    @Override
    public void addLibrary(Library library) {
        this.libraries.add(library);
    }

    @Override
    public void addLibraries(Library... libraries) {
        for (Library library : libraries) {
            this.addLibrary(library);
        }
    }

    @Override
    public Collection<Library> getLibraries() {
        return this.libraries;
    }

    @Override
    public void loadLibraries() {
        libraries.forEach(this::loadLibrary);
    }
}

File: core/src/main/kotlin/me/adrigamer2950/adriapi/api/library/manager/LibraryManagerImpl.kt
package me.adrigamer2950.adriapi.api.library.manager

import com.alessiodp.libby.Library
import com.alessiodp.libby.StandaloneLibraryManager
import com.alessiodp.libby.logging.adapters.JDKLogAdapter
import org.bukkit.Bukkit
import org.bukkit.plugin.Plugin
import org.jetbrains.annotations.ApiStatus.Internal
import java.io.File

@Internal
class LibraryManagerImpl internal constructor(override val plugin: Plugin, folderName: String = plugin.name) : StandaloneLibraryManager(
    JDKLogAdapter(plugin.logger), File(
        Bukkit.getPluginsFolder(), folderName
    ).toPath(), "lib"
), LibraryManager {

    override val libraries: MutableSet<Library> = HashSet()

    init {
        this.addMavenCentral()
    }

    override fun addLibrary(library: Library) {
        this.libraries.add(library)
    }

    override fun addLibraries(vararg libraries: Library) {
        libraries.forEach { this.addLibrary(it) }
    }

    override fun loadLibraries(vararg libraries: Library) {
        libraries.forEach { this.loadLibrary(it) }
    }

    override fun loadLibraries() {
        libraries.forEach { this.loadLibrary(it) }

        libraries.clear()
    }
}
